### PR TITLE
UML diagrams for DG use

### PR DESCRIPTION
**Pull request by jiale-c**
The purpose of this PR is to edit the developer guide for adding diagrams for piechartwindow, summarise, Ui.

This PR is an initial implementation and would require vetting of diagrams by my fellow teammates

This PR is not dependent on any other PRs.

### CODE REVIEW

File: docs/diagrams/SummariseClassDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+skinparam classAttributeIconSize 0
+hide circle
+
+abstract class "{abstract}\nCommand" {
+    + {abstract} execute(model : Model) : CommandResult
+}
+
+interface Model <<interface>> {
+    + addPerson(Person person) : void
+    + hasPerson(Person person) : boolean
```
Comment (cwq2326) [inline docs/diagrams/SummariseClassDiagram.puml:12]: Similiar to the class diagram for pie chart window

For function that returns nothing, I think you can omit the void part
```suggestion
    + addPerson(Person)
    + hasPerson(Person) : boolean
```

File: docs/diagrams/PieChartWindowClassDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    - {static} logger : Logger
+    - {static} FXML : String
+    - covidStatsByBlockData : TreeMap
+    - positiveStatsByFacultyData : TreeMap
+    - pieChart : PieChart
+    - charts : VBox
+    - chartScene : Scene
+
+    /' Methods '/
+    + PieChartWindow(root : Stage)
```
Comment (cwq2326) [inline docs/diagrams/PieChartWindowClassDiagram.puml:32]: I think you can omit the variable name of the parameter for root
```suggestion
    + PieChartWindow(root)
```

File: docs/diagrams/SummariseFeatureActivityDiagram.puml [hunk 3]
```diff
@@ -1,19 +1,22 @@
 @startuml
 'https://plantuml.com/activity-diagram-beta
 
-(*)--> "User inputs Summarise Command without any tags"
--right-> "Tracey collates total number of students with covid"
---> "Tracey collates number of students in each\nhall block under each covid status category"
--left-> "Tracey collates number of\ncovid positive students in each faculty"
+start
+:User input summarise command;
```
Comment (cwq2326) [inline docs/diagrams/SummariseFeatureActivityDiagram.puml:5]: Perhaps could add in the part whereby the user put in an irrelevant parameter for this command

File: docs/diagrams/UiClassDiagram.puml [hunk 4]
```diff
@@ -15,6 +15,8 @@ Class PersonListPanel
 Class PersonCard
 Class StatusBarFooter
 Class CommandBox
+Class EmailWindow
+Class PieChartWindow
 }
```
Comment (cwq2326) [inline docs/diagrams/UiClassDiagram.puml:21]: Perhaps could include the buttons (resize, copy email) as part of the UI class diagram

**Review by cwq2326 — CHANGES_REQUESTED** [review]
Just some suggestions for changes for the class and activity diagrams.

File: docs/diagrams/SummariseClassDiagram.puml (hunk 1 above)
Comment (tanchinkiat99) [reply docs/diagrams/SummariseClassDiagram.puml:12]: I agree with this change.

Comment (jiale-c) [reply docs/diagrams/SummariseClassDiagram.puml]: Thanks for the suggestions. I have made the amendments.

File: docs/diagrams/PieChartWindowClassDiagram.puml (hunk 2 above)
Comment (jiale-c) [reply docs/diagrams/PieChartWindowClassDiagram.puml]: Thanks for the suggestions. I have made the amendments.

File: docs/diagrams/SummariseFeatureActivityDiagram.puml (hunk 3 above)
Comment (jiale-c) [reply docs/diagrams/SummariseFeatureActivityDiagram.puml:5]: Definitely. Thanks for highlighting this as I have missed it.

File: docs/diagrams/PieChartWindowClassDiagram.puml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    /' Fields '/
+    - {static} logger : Logger
+    - {static} FXML : String
+    - covidStatsByBlockData : TreeMap
+    - positiveStatsByFacultyData : TreeMap
+    - pieChart : PieChart
+    - charts : VBox
+    - chartScene : Scene
+
+    /' Methods '/
```
Comment (DannyDakota) [inline docs/diagrams/PieChartWindowClassDiagram.puml:31]: I think you should consider including the return type for the methods 👍

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 6]
```diff
@@ -278,17 +278,13 @@ private void handleExit() {
...
+            if (pieChartWindow.isShowing()) {
                 pieChartWindow.hide();
-                pieChartWindow = new PieChartWindow();
-                pieChartWindow.show();
-                logger.info("Pie chart window already showing, proceeding to reopen it!");
+                logger.info("Pie chart window is not yet initialised or not showing!");
             }
+            pieChartWindow = new PieChartWindow();
+            pieChartWindow.execute();
+            pieChartWindow.show();
```
Comment (DannyDakota) [inline src/main/java/seedu/address/ui/MainWindow.java:287]: Great method of removing the if branch! well done! 😄

**Review by DannyDakota — CHANGES_REQUESTED** [review]
Just addition of Return types for the methods, other than that and what wen quan has mentioned, everything else looks good!

**Review by cwq2326 — APPROVED** [review]
Thanks for making the necessary changes, LGTM 💯

**Review by DannyDakota — APPROVED** [review]
LGTM! Thank you for your hard work! 🎆

**Pull request merged**